src/main/java/com/asu/seatr/handlers/CourseHandler.java

```diff
@@ -72,4 +74,53 @@ public static void updateCourseByExternalID(String external_id,Course course){
 		session.getTransaction().commit();
 		session.close();
 	}
+	
+	public static List<Integer> getInternalCourseList(Set<String> courseIdList)
+	{
+		if(courseIdList == null || courseIdList.isEmpty())
+		{
+			return null;
+		}
+		try
+		{
+		SessionFactory sf = HibernateUtil.getSessionFactory();
+		Session session = sf.openSession();
+		session.beginTransaction();
+		String hql = "select c.id from Course c where c.external_id in :courseIdList";

```

Comment:
Because, we are not using single equality condition. we are checking the whole list